### ISSUE TITLE
No break set in switch statements

### ISSUE THREAD

**Issue by mneil** [HeliosInteractive/infinity#1]
Break is not being called on switch statements. If you do not call break the expression is evaluated against all cases. Not only is this a performance issue — but if you specify a default it will break (pun intended) your code. Example:

```
switch(true){case true: console.log('true');default: console.log('false');}
```
Throw that in your console and run it. 

Always call break in a switch. 

My favorite way to avoid this pitfall is to not use switches, ever ;)

**Comment by mneil** [comment]
I added a new branch, es6-perf with more code moved to es6 syntax for brevity and fixed missing break statements. I also added a few minor performance tweaks like removing regex splitting of class/id name for element selector and used substring instead and also allowed passing in DOM elements directly instead of strings.

**Comment by adammcfadden** [comment]
Solved by #2
